ui/perfherder/compare/CompareTable.jsx

```diff
@@ -1,19 +1,119 @@
 import React from 'react';
-import { Button, Table } from 'reactstrap';
+import { Badge, Button, Table } from 'reactstrap';
 import PropTypes from 'prop-types';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faHashtag } from '@fortawesome/free-solid-svg-icons';
+import {
+  faHashtag,
+  faSort,
+  faSortDown,
+  faSortUp,
+} from '@fortawesome/free-solid-svg-icons';
 
 import { getHashBasedId } from '../helpers';
 import { hashFunction } from '../../helpers/utils';
 
 import RetriggerButton from './RetriggerButton';
 import CompareTableRow from './CompareTableRow';
 
-export default class CompareTable extends React.PureComponent {
+export default class CompareTable extends React.Component {
+  sortTypes = {
+    ascending: {
+      icon: faSortUp,
+      sortByString: (value) => (a, b) => a[value].localeCompare(b[value]),
+      sortByValue: (value) => (a, b) => a[value] - b[value],
+    },
+    descending: {
+      icon: faSortDown,
+      sortByString: (value) => (a, b) => b[value].localeCompare(a[value]),
+      sortByValue: (value) => (a, b) => b[value] - a[value],
+    },
+    default: { icon: faSort },
+  };
+
+  constructor(props) {
+    super(props);
+    const { data } = this.props;
+
+    this.state = {
+      data,
+      columns: [
+        { name: 'Test name', sortValue: 'name', currentSort: 'default' },
+        { name: 'Base', sortValue: 'originalValue', currentSort: 'default' },
+        { name: 'Comparison' },
+        { name: 'New', sortValue: 'newValue', currentSort: 'default' },
+        { name: 'Delta', sortValue: 'deltaPercentage', currentSort: 'default' },
+        {
+          name: 'Magnitude of Difference',
+          sortValue: 'magnitude',
+          currentSort: 'default',
+        },
+        { name: 'Confidence', sortValue: 'confidence', currentSort: 'default' },
+      ],
+    };
+  }
+
+  componentDidUpdate = (prevProps) => {
+    const { data } = this.props;
+    let { columns } = this.state;
+
+    if (data !== prevProps.data) {
+      columns = columns.map((item) => {
+        item.currentSort = 'default';
+        return item;
+      });
+      this.setNewState({ data, columns });
+    }
+  };
+
+  setNewState = (newState) => this.setState(newState);
```

Comment:
I don't see any value in having a dedicated method like this one.

Reply:
I tried to avoid disabling this eslint rule `eslint-disable-next-line react/no-did-update-set-state`